File: algo-pig-latin/js/pigLatin.js
exports.translate = function(word) {
    output = '';
    word.split(' ').forEach((w)=> {
        i = 0;
        is_q = false;
        while ((i < w.length) && ((/[^aeiou]/.test(w[i])) || (w[i] == 'u' && is_q))) {
            (w[i++] == 'q') ? is_q = true : is_q = false;
        }
        output += w.slice(i) + w.slice(0, i) + 'ay ';
    });
    return output.slice(0, -1);
};

File: algo-pig-latin/python/pig_latin.py
def is_vowel(c):
  return c in {'a', 'e', 'i', 'o', 'u'}

def translate(word_or_phrase):
  output = ''
  word_arr = word_or_phrase.split(' ')
  for word in word_arr:
    i = 0
    is_q = False
    while i < len(word) and ((not is_vowel(word[i])) or (word[i] == 'u' and is_q)):
      is_q = word[i] == 'q'
      i += 1
    output += word[i:] + word[:i] + 'ay '

  return output[:-1]